src/chrome/content/ff-overlay.js

```diff
@@ -85,30 +85,37 @@ if (typeof SL_Overlay === "undefined") {
                 .addEventListener("popupshowing", (event) => {
                     this.showFirefoxContextMenu(event);
                 }, false);
-            DownloadManager.connectToNas();
         };
 
         this.onMenuItemLinkCommand = (event) => {
-            if (DownloadManager.isConnected) {
-                window.open(DownloadManager.urlToConnect + "/webman/index.cgi?launchApp=SYNO.SDS.DownloadStation.Application", "Diskstation", this.strWindowFeatures);
-            } else {
-                Notification.show("No Connection");
-            }
+            DownloadManager.connectAndRun(() => {
+                if (DownloadManager.isConnected) {
+                    window.open(DownloadManager.urlToConnect + "/webman/index.cgi?launchApp=SYNO.SDS.DownloadStation.Application", "Diskstation", this.strWindowFeatures);
+                } else {
+                    Notification.show("No Connection");
+                }
+            });
         };
 
         this.onMenuItemCommand = (event) => {
-            if (DownloadManager.isConnected) {
-                DownloadManager.transferToNas(gContextMenu.linkURL);
-            } else {
-                Notification.show("No Connection");
-            }
+            let linkURL = gContextMenu.linkURL;
+            DownloadManager.connectAndRun(() => {
```

Comment:
calling a function off the same class just to call a another function of this class seams wrong in terms of the OOP. So i would suggest to implement the connecting in DownloadManager and keep the interface to DownloadManager as simple as possible. 
Sound ok to you ?